Packages/com.unity.inputsystem/InputSystem/SourceGenerators/Source~/SourceGenerators/GlobalInputActionsSourceGenerator.cs

```diff
@@ -0,0 +1,372 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Unity.InputSystem.SourceGenerators
+{
+    [Generator]
+    public class GlobalInputActionsSourceGenerator : ISourceGenerator
+    {
+        //@TODO: This needs a JSON file
+        //@TODO: This is pointing to the template at the moment (The active user file is InputManager.asset, which is not JSON)
+        private const string GlobalActionsAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsTemplate.inputactions";
+
+        // Entry assembly is null in VS IDE
+        // At build time, it is `csc` or `VBCSCompiler`
+        private static bool IsBuildTime => Assembly.GetEntryAssembly() != null;
+
+        public void Initialize(GeneratorInitializationContext context)
+        {
+            if (IsBuildTime)
+                context.RegisterForSyntaxNotifications(() => new InputActionsSyntaxReceiver());
+        }
+
+// Code formatter does not like raw literals
+//*begin-nonstandard-formatting*
+        public void Execute(GeneratorExecutionContext context)
+        {
+            // only run at build time
+            if (!IsBuildTime)
+                return;
+
+            // This should never be caught
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            var additionalFile = context.AdditionalFiles.FirstOrDefault();
+            if (additionalFile == null)
+            {
+                context.ReportDiagnostic(Diagnostic.Create(
+                    new DiagnosticDescriptor("ISGEN001", "Additional files not specified.", "", "InputSystemSourceGenerator", DiagnosticSeverity.Error, true, "No additional files were specified."), null));
+                return;
+            }
+
+            var projectPath = additionalFile.GetText().ToString();
+            var globalActionsAssetPath = Path.Combine(projectPath, GlobalActionsAssetPath);
+            if (!File.Exists(globalActionsAssetPath))
+            {
+                Console.Error.Write($"Global input actions asset file not found at '{globalActionsAssetPath}'.");
+                return;
+            }
+
+            using var fileStream = new FileStream(globalActionsAssetPath, FileMode.Open);
```

Comment:
This seems to just be processing the template file and note the real data in project wide input actions ?

Reply:
Yes, this was noted in the PR description but now fixed.